lib/Backend/SccLiveness.cpp

```diff
@@ -53,24 +53,30 @@ SCCLiveness::Build()
             AssertMsg(instr->IsBranchInstr(), "Loop tail should be a branchInstr");
             AssertMsg(instr->AsBranchInstr()->IsLoopTail(this->func), "Loop tail not marked correctly");
 
-            FOREACH_SLIST_ENTRY(Lifetime *, lifetime, this->curLoop->regAlloc.extendedLifetime)
+            Loop *loop = this->curLoop;
+            while (loop && loop->regAlloc.loopEnd == this->curLoop->regAlloc.loopEnd)
```

Comment:
>== [](start = 50, length = 2)

Shouldn't this be <= or is it guaranteed to have been extended by now?  Would still be safer...